Rubberduck.Inspections/Concrete/UnreachableCaseInspection/UnreachableCaseInspection.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Antlr4.Runtime;
+using Antlr4.Runtime.Misc;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Resources;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Inspections.Concrete.UnreachableCaseInspection
+{
+    public sealed class UnreachableCaseInspection : ParseTreeInspectionBase
+    {
+        private IUCIParseTreeValueVisitorFactory _parseTreeVisitorFactory;
+        private IUnreachableCaseInspectionSelectStmtFactory _selectStmtFactory;
+        private IUCIValueFactory _valueFactory;
+        private IUCIParseTreeValueVisitor _parseTreeValueVisitor;
+        private enum CaseInpectionResult { Unreachable, MismatchType, CaseElse };
+
+        private static Dictionary<CaseInpectionResult, string> ResultMessages = new Dictionary<CaseInpectionResult, string>()
+        {
+            [CaseInpectionResult.Unreachable] = InspectionsUI.UnreachableCaseInspection_Unreachable,
+            [CaseInpectionResult.MismatchType] = InspectionsUI.UnreachableCaseInspection_TypeMismatch,
+            [CaseInpectionResult.CaseElse] = InspectionsUI.UnreachableCaseInspection_CaseElse
+        };
+
+        public UnreachableCaseInspection(RubberduckParserState state) : base(state)
+        {
+            //TODO_Question: IUnreachableCaseInspectionFactoryFactory - candidate for IoCInstaller?
+            var factoriesFactory = new UnreachableCaseInspectionFactoryFactory();
```

Comment:
Yes. IMO, They are. I think we want to use them as much as possible to avoid manual maintenance when the DI needs to be updated for whatever.